packages/twenty-front/src/modules/apollo/optimistic-effect/utils/isCachedObjectConnection.ts

```diff
@@ -0,0 +1,21 @@
+import { StoreValue } from '@apollo/client';
+import { z } from 'zod';
+
+import { CachedObjectRecordConnection } from '@/apollo/types/CachedObjectRecordConnection';
+import { capitalize } from '~/utils/string/capitalize';
+
+export const isCachedObjectConnection = (
```

Comment:
For a cached connection to be a reference, it would need to have a unique id so Apollo can identify it and reference it in the cache, which is not the case. Same for cached edges. In the cache currently, connections and edges are never references. Only records (Company, People...) are refs.

Reply:
Could we maintain only a single type then ? I'm not sure we need both then

Reply:
We do need both because `CachedObjectRecordConnection` contains an array of `CachedObjectRecordEdge`s, which is:

```
export type CachedObjectRecordEdge = Omit<ObjectRecordEdge, 'node'> & {
  node: Reference;
};
```

So `CachedObjectRecordConnection` contains References to cached records, `ObjectRecordConnection` contains records.